Game/RainMeadow.CreatureHooks.cs

```diff
@@ -33,11 +33,39 @@ private void CreatureHooks()
             On.EggBugGraphics.Update += EggBugGraphics_Update;
             On.BigSpiderGraphics.Update += BigSpiderGraphics_Update;
 
+            On.SSOracleBehavior.NewAction += SSOracleBehavior_NewAction;
+
             On.EggBug.DropEggs += EggBug_DropEggs;
             On.Vulture.DropMask += Vulture_DropMask;
             On.BigSpider.BabyPuff += BigSpider_BabyPuff;
         }
 
+        private void SSOracleBehavior_NewAction(On.SSOracleBehavior.orig_NewAction orig, SSOracleBehavior self, SSOracleBehavior.Action nextAction)
+        {
+            if (OnlineManager.lobby != null && self.oracle.abstractPhysicalObject.GetOnlineObject(out var opo))
+            {
+                if (opo.roomSession.isOwner && opo.isMine)
+                { //we own the entity xaxax
+                    foreach (var player in OnlineManager.players)
+                    {
+                        if (!player.isMe)
+                        {
+                            player.InvokeOnceRPC(opo.TriggerOracleAction, nextAction);
```

Comment:
also wait this should be state not RPC

sync the action in the realizedoraclestate and set accordingly in readto like how we do moon's revival